src/coreclr/jit/simd.h

```diff
@@ -520,6 +527,16 @@ TBase EvaluateBinaryScalarSpecialized(genTreeOps oper, TBase arg0, TBase arg1)
             return arg0 ^ arg1;
         }
 
+        case GT_EQ:
+        {
+            return arg0 == arg1 ? GetAllBitsSetScalar<TBase>() : 0;
+        }
+
+        case GT_NE:
+        {
+            return arg0 != arg1 ? GetAllBitsSetScalar<TBase>() : 0;
+        }
```

Comment:
This is going to be incorrect for `float`/`double` since it will do a bitwise comparison.

You may need to special-case in `EvaluateBinaryScalarSpecialized<float>` before it calls this main method.

Reply:
You might also be able to do it in `EvaluateBinaryScalar` instead but special-case `GetAllBitsSetScalar<float>()`, which is probably simpler overall.

Reply:
This path is never taken for floating point, I added asserts just in case. I was mostly interested in non-fp cases and these can be added if needed

Reply:
Any reason we aren’t covering floating-point as well? I imagine this might get hits in ImageSharp or PDN.

Potentially in other apps, games, or graphics processing, etc